Occasional `docker.errors.NotFound` during testing

> [!note]
> If someone else encounters this, I'd appreciate full tracebacks/logs/commands to help debugging.
> As far as workarounds, you should probably simply be able to start your run again; this seems to happen only occasionally.

```
==================================== ERRORS ====================================
___________________ ERROR at teardown of test_env_with_hook ____________________

self = <docker.api.client.APIClient object at 0x7f1ef9b645e0>
response = <Response [404]>

    def _raise_for_status(self, response):
        """Raises stored :class:`APIError`, if one occurred."""
        try:
>           response.raise_for_status()

/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/docker/api/client.py:275: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Response [404]>

    def raise_for_status(self):
        """Raises :class:`HTTPError`, if one occurred."""
    
        http_error_msg = ""
        if isinstance(self.reason, bytes):
            # We attempt to decode utf-8 first because some servers
            # choose to localize their reason strings. If the string
    resp = self.client.api.inspect_container(container_id)
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/docker/utils/decorators.py:19: in wrapped
    return f(self, resource_id, *args, **kwargs)
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/docker/api/container.py:793: in inspect_container
    return self._result(
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/docker/api/client.py:281: in _result
    self._raise_for_status(response)
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/docker/api/client.py:277: in _raise_for_status
    raise create_api_error_from_http_exception(e) from e
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

e = HTTPError('404 Client Error: Not Found for url: http+docker://localhost/v1.43/containers/test-container-134245890345098/json')

    def create_api_error_from_http_exception(e):
        """
        Create a suitable APIError from requests.exceptions.HTTPError.
        """
        response = e.response
        try:
            explanation = response.json()['message']
        except ValueError:
            explanation = (response.text or '').strip()
        cls = APIError
        if response.status_code == 404:
            explanation_msg = (explanation or '').lower()
            if any(fragment in explanation_msg
                   for fragment in _image_not_found_explanation_fragments):
                cls = ImageNotFound
            else:
                cls = NotFound
>       raise cls(e, response=response, explanation=explanation) from e
E       docker.errors.NotFound: 404 Client Error for http+docker://localhost/v1.43/containers/test-container-134245890345098/json: Not Found ("No such container: test-container-134245890345098")

/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/site-packages/docker/errors.py:39: NotFound
```

--- a/sweagent/environment/swe_env.py
+++ b/sweagent/environment/swe_env.py
@@ -2,6 +2,7 @@ from __future__ import annotations
 
 import datetime
 import hashlib
+import json
 import logging
 import os
 import random
@@ -21,6 +22,7 @@ from simple_parsing.helpers.serialization.serializable import FrozenSerializable
 from swebench import MAP_VERSION_TO_INSTALL, get_environment_yml, get_requirements
 
 import docker
+import docker.errors
 import docker.models.containers
 from sweagent import REPO_ROOT
 from sweagent.environment.utils import (
@@ -570,12 +572,23 @@ class SWEEnv(gym.Env):
             else:
                 msg = "Unknown docker exception occurred. Are you sure docker is running?"
             raise RuntimeError(msg) from e
-        try:
-            self.container_obj = client.containers.get(self.container_name)
-        except docker.errors.NotFound:
-            logger.debug("Couldn't find container. Let's wait and retry.")
-            time.sleep(3)
-            self.container_obj = client.containers.get(self.container_name)
+        t0 = time.time()
+        self.container_obj = None
+        while time.time() - t0 < 60:
+            try:
+                self.container_obj = client.containers.get(self.container_name)
+            except docker.errors.NotFound:
+                logger.debug("Couldn't find container. Let's wait and retry.")
+                time.sleep(1)
+            else:
+                break
+        else:
+            print(f"{self.persistent=}")
+            available_containers = client.containers.list(all=True)
+            available_containers_info = json.dumps([str(c.attrs) for c in available_containers], indent=2)
+            print(available_containers_info)
+            msg = "Failed to get container object."
+            raise RuntimeError(msg)
         self.logger.info("🌱 Environment Initialized")
 
     def _init_scripts(self):

